src/presentation/api/controller/services.go

```diff
@@ -0,0 +1,248 @@
+// @Security     Bearer
+// @Accept       json
+// @Produce      json
+// @Success      200 {array} entity.InstallableService
+// @Router       /v1/services/installables/ [get]
+func (controller *ServicesController) ReadInstallables(c echo.Context) error {
+	return apiHelper.ServiceResponseWrapper(
+		c, controller.serviceService.ReadInstallables(),
+	)
+}
+
+func parseRawPortBindings(bindings []interface{}) []string {
+	rawPortBindings := []string{}
+	for _, rawPortBinding := range bindings {
+		rawPortBindingMap, assertOk := rawPortBinding.(map[string]interface{})
+		if !assertOk {
+			continue
+		}
+
+		rawPortStr, assertOk := rawPortBindingMap["port"].(string)
+		if !assertOk {
+			rawPortFloat, assertOk := rawPortBindingMap["port"].(float64)
+			if !assertOk {
+				continue
+			}
+			rawPortStr = strconv.FormatFloat(rawPortFloat, 'f', -1, 64)
+		}
+
+		rawProtocolStr, assertOk := rawPortBindingMap["protocol"].(string)
+		if !assertOk {
+			continue
+		}
+
+		rawPortBinding = rawPortStr + "/" + rawProtocolStr
+		rawPortBindings = append(rawPortBindings, rawPortBinding.(string))
+	}
+
+	return rawPortBindings
+}
+
+// CreateInstallableService godoc
+// @Summary      CreateInstallableService
+// @Description  Install a new installable service.
+// @Tags         services
+// @Accept       json
+// @Produce      json
+// @Security     Bearer
+// @Param        createInstallableServiceDto	body dto.CreateInstallableService	true	"Only name is required.<br />If version is not provided, it will be 'lts'.<br />If portBindings is not provided, it wil be default service port bindings.<br />If autoCreateMapping is not provided, it will be 'true'."
+// @Success      201 {object} object{} "InstallableServiceCreated"
+// @Router       /v1/services/installables/ [post]
+func (controller *ServicesController) CreateInstallable(c echo.Context) error {
+	requestBody, err := apiHelper.ReadRequestBody(c)
+	if err != nil {
+		return err
+	}
+
+	rawEnvsSlice := []string{}
+	if requestBody["envs"] != nil {
+		for _, rawEnv := range requestBody["envs"].([]interface{}) {
+			rawEnvsSlice = append(rawEnvsSlice, rawEnv.(string))
+		}
+	}
+	requestBody["envs"] = rawEnvsSlice
+
+	rawPortBindings := []string{}
+	if requestBody["portBindings"] != nil {
+		rawPortBindings = parseRawPortBindings(
+			requestBody["portBindings"].([]interface{}),
+		)
+	}
+	requestBody["portBindings"] = rawPortBindings
+
+	if requestBody["timeoutStartSecs"] != nil {
+		requestBody["timeoutStartSecs"], err = voHelper.InterfaceToUint(
+			requestBody["timeoutStartSecs"],
+		)
+		if err != nil {
+			return apiHelper.ResponseWrapper(
+				c, http.StatusBadRequest, "TimeoutStartSecsMustBeUint",
+			)
+		}
+	}
+
+	if requestBody["maxStartRetries"] != nil {
+		requestBody["maxStartRetries"], err = voHelper.InterfaceToUint(
+			requestBody["maxStartRetries"],
+		)
+		if err != nil {
+			return apiHelper.ResponseWrapper(
+				c, http.StatusBadRequest, "MaxStartRetriesMustBeUint",
+			)
+		}
+	}
```

Comment:
Isn't this useless since you can use the interface to uint on the service side?

Reply:
I thought I would end up doing the parsing twice by putting this in the service since the **CLI** already guarantees the typing for me.

Anyway, I moved it to the service to reduce the lines here.